include/Reader/readerir.h

```diff
@@ -1600,6 +1604,33 @@ class GenIR : public ReaderBase {
                                           uint64_t ValueHandle, llvm::Type *Ty,
                                           llvm::StringRef Name,
                                           bool IsConstant);
+  std::string appendClassNameAsString(CORINFO_CLASS_HANDLE Class, bool IncludeNamespace,
+                                      bool FullInst, bool IncludeAssembly);
+  IRNode *vectorAdd(IRNode *Vector1, IRNode *Vector2);
+  IRNode *vectorSub(IRNode *Vector1, IRNode *Vector2);
+  IRNode *vectorMul(IRNode *Vector1, IRNode *Vector2);
+  IRNode *vectorDiv(IRNode *Vector1, IRNode *Vector2);
+  IRNode *vectorEqual(IRNode *Vector1, IRNode *Vector2);
+  IRNode *vectorNotEqual(IRNode *Vector1, IRNode *Vector2);
+  IRNode *vectorAbs(IRNode *Vector);
+  IRNode *vectorSqrt(IRNode *Vector);
+
+  bool checkVectorType(IRNode * Arg);
+
+  bool checkVectorSignature(std::vector<IRNode*> Args, std::vector<llvm::Type*> Types);
+  IRNode * vectorCtor(CORINFO_CLASS_HANDLE Class, IRNode * This, std::vector<IRNode*> Args);
+  IRNode * vectorCtorFromOne(int VectorSize, IRNode * This, std::vector<IRNode*> Args);
+  IRNode * vectorCtorFromFloats(int VectorSize, IRNode * This, std::vector<IRNode*> Args);
```

Comment:
I'd guess that most of these uses of `std::vector<IRNode*>` can be replaced with uses of `llvm::ArrayRef<IRNode*>`